addon/mixins/service-mixin.js

```diff
@@ -176,20 +176,38 @@ export default Mixin.create({
    * Wrap the options passed to rest-js with auth info and use ember-fetch.
    */
   addOptions (args, portalOpts) {
-    // always use ember-fetch
-    args.fetch = fetch;
-
-    // force rest-js to use whatever the fuck EAPS thinks is the right portal URL
-    args.portal = this.getPortalRestUrl(portalOpts);
-
+    let authMgr = this.get('session.authMgr');
+    // ---------------------------------------------------
+    // TODO: just a test - this should be done in Torii
+    let correctPortalRestUrl = this.getPortalRestUrl();
+    if (authMgr.portal !== correctPortalRestUrl) {
+      debug(`AuthMgr.portal: ${authMgr.portal}`);
+      debug(`session.portalHostname: ${this.get('session.portalHostname')}`);
+      this.set('session.authMgr.portal', correctPortalRestUrl);
+    }
+    // ---------------------------------------------------
     // exactly the same token logic as in requestUrl() above
-    const token = portalOpts ? portalOpts.token : this.get('session.token');
-    if (token) {
-      if (!args.params) {
-        args.params = {};
+    if (portalOpts && portalOpts.token) {
+      // for some reason we REALLY want to use this token...
+      // so we construct a minimalist IAuthenticationManger
+      let portalToUse = portalOpts.portalHostname || this.get('session.portalHostname');
```

Comment:
These 3 lines should be replaced w/ `let portalToUse = this.getPortalRestUrl(portalOpts);` Let it do it's thing (which includes [fixing the url for enterprise environments](https://github.com/Esri/ember-arcgis-portal-services/blob/v1.5.0/addon/mixins/service-mixin.js#L54-L58)).